examples/multi_gpu/papers100m_gcn.py

```diff
@@ -157,7 +157,15 @@ def run_train(rank, data, world_size, model, epochs, batch_size, fan_out,
 
     args = parser.parse_args()
     wall_clock_start = time.perf_counter()
-
+    if args.n_devices == -1:
+        world_size = torch.cuda.device_count()
+    else:
+        world_size = args.n_devices
+    import psutil
+    if (psutil.virtual_memory().total / (1024**3)) < 190 + 200 * world_size:
+        print("Warning: may not have enough RAM to use this many GPUs.")
+        print("Consider upgrading RAM or using less GPUs if an error occurs.")
+    print('Let\'s use', world_size, 'GPUs!')
```

Comment:
Can we have a specific memory requirement, e.g., "requires xxx GB", printed out, too?